app/src/test/java/org/mozilla/fenix/components/metrics/AppStartupTelemetryTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.mozilla.fenix.components.metrics
+
+import android.content.Intent
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.verify
+import mozilla.components.support.utils.toSafeIntent
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.LINK
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.CUSTOM_TAB
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.APP_ICON
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.UNKNOWN
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.COLD
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.WARM
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.HOT
+
+class AppStartupTelemetryTest {
+
+    @RelaxedMockK
+    private lateinit var metricController: MetricController
+    @RelaxedMockK
```

Comment:
nit: if you can easily avoid relaxing a mock, you should try. Non-relaxed mocks help you ensure you're mocking all the methods that are called and 1) are testing all the possible code paths and 2) aren't writing in states that would never happen in the code (where unit is returned instead of a real value).

Relaxing is best used e.g. when an object is passed in as an argument and it might be queried but it's values do not impact the final result (which does assume some knowledge about the implementation, to be fair, and that's an anti-pattern).